libs/vkd3d/state.c

```diff
@@ -4082,8 +4084,25 @@ VkPipeline d3d12_pipeline_state_create_pipeline_variant(struct d3d12_pipeline_st
     vp_desc.scissorCount = key && !key->dynamic_viewport ? max(key->viewport_count, 1) : 0;
     vp_desc.pScissors = NULL;
 
+    rtv_active_mask = key ? key->rtv_active_mask : graphics->rtv_active_mask;
+    for (i = 0; i < D3D12_SIMULTANEOUS_RENDER_TARGET_COUNT; i++)
+    {
+        if (rtv_active_mask & (1u << i))
+            rtv_formats[i] = graphics->rtv_formats[i];
+        else
+            rtv_formats[i] = VK_FORMAT_UNDEFINED;
+    }
+
+    rendering_info.sType = VK_STRUCTURE_TYPE_PIPELINE_RENDERING_CREATE_INFO_KHR;
+    rendering_info.pNext = NULL;
+    rendering_info.viewMask = 0;
+    rendering_info.colorAttachmentCount = D3D12_SIMULTANEOUS_RENDER_TARGET_COUNT;
```

Comment:
Shouldn't colorAttachmentCount be restricted to findMSB(rtv_active_mask) + 1?

Reply:
do we have a way to find the msb?

Reply:
Shouldn't make any difference, but I guess we can do that.

Reply:
Alternatively, NumRenderTargets could be used, might be more instructive.

Reply:
fixed.